### PR TITLE
New Field - termination_date

### PR DESCRIPTION
**Pull request by mplsmitch**
Adds a new field, termination_date, to system_information.json to notify data consumers when feeds will be discontinued.

 

#### **What problem does your proposal solve? Please begin with the relevant issue number. If there is no existing issue, please also describe alternative solutions you have considered.**

The specification lacks a way to notify data consumers when a feed is shutting down permanently and will no longer be available. This has come up a number of times in the past but we have yet to formalize a best practice for feed end of life. Relevant issue with further discussion is #492.

#### **What is the proposal?**
Add a field `termination_date` to `system_information.json` that contains the date when the feed will go off line. Define best practice for communicating with data consumers about feed end of life.

#### **Is this a breaking change?**
- [ ] Yes 
- [x] No
- [ ] Unsure

#### **Which files are affected by this change?**
gbfs.md > system_information.json

### CODE REVIEW

**Comment by mplsmitch** [comment]
I hereby call a vote on this proposal. Voting will be open for 10 full calendar days until 11:59PM UTC on May 4th.
Please vote for or against the proposal, and include the organization for which you are voting in your comment.
Please note if you can commit to implementing the proposal.

A few notes on voting:

- Anyone may vote on a proposal. You do not need to be a member of MobilityData.
- A vote passes if there is unanimous consensus with at least 3 votes in favor.
- At least one of the 3 votes MUST be from a GBFS producer and at least one MUST be from a GBFS consumer.
- Votes against a proposal can stop it from passing if they provide a specific reason for voting against and contain actionable feedback.
- If you represent an organization, only one vote per organization will be counted. 

CC: @mwillmott

**Comment by mwillmott** [comment]
Note that the original comment refers to `station_information.json` but i believe it is meant to be `system_information.json` as reflected in the actual changes made.

On that basis, Lyft (operator) votes for this proposal.

cc @ArashMansouri

**Comment by mplsmitch** [comment]
> Note that the original comment refers to `station_information.json` but i believe it is meant to be `system_information.json` as reflected in the actual changes made.
> 
@mwillmott  thanks for catching that, I've corrected it

**Comment by futuretap** [comment]
+1 from FutureTap / Where To? (consumer)

**Comment by cmonagle** [comment]
+1 from Transit

**Comment by benwedge** [comment]
+1 from Joyride

**Comment by fbouchPBSC** [comment]
+1 from PBSC

**Comment by josee-sabourin** [comment]
Voting on this PR closes in 2 calendar days. Please vote for or against the proposal, and include the organization for which you are voting in your comment. Please note if you can commit to implementing the proposal.

**Comment by ezmckinn** [comment]
+1 Superpedestrian supports this proposal.

**Comment by josee-sabourin** [comment]
This vote has now closed, and it passes!

Votes in favor:
FutureTap / Where To? (consumer)
Transit (consumer)
Joyride (producer)
PBSC (producer)
Superpedestrian (producer)
 
There were no votes against.

**Review by richfab — APPROVED** [review]
LGTM

File: gbfs.md [hunk 1]
```diff
@@ -397,6 +399,7 @@ Field Name | REQUIRED | Type | Defines
 `url` | OPTIONAL | URL | The URL of the vehicle share system.
 `purchase_url` | OPTIONAL | URL | URL where a customer can purchase a membership.
 `start_date` | OPTIONAL | Date | Date that the system began operations.
+`termination_date` | OPTIONAL | Date | Date after which this data source will no longer be available to consuming applications.<br><br>This OPTIONAL field SHOULD be used to notify 3rd party data consumers when a service is planning a permanent (non-seasonal) shutdown. Publishers SHOULD include this date in their feeds as soon as they know of an impending shutdown. Publishers SHOULD continue to publish feeds for 30 days following a permanent shutdown after which they SHOULD return a helpful http status code and text, for example `410 service no longer available ...` for all feed endpoint URLs.
```
Comment (richfab) [inline gbfs.md]: Added in v3.0-RC2

```suggestion
`termination_date` <br/>*(added in v3.0-RC2)* | OPTIONAL | Date | Date after which this data source will no longer be available to consuming applications.<br><br>This OPTIONAL field SHOULD be used to notify 3rd party data consumers when a service is planning a permanent (non-seasonal) shutdown. Publishers SHOULD include this date in their feeds as soon as they know of an impending shutdown. Publishers SHOULD continue to publish feeds for 30 days following a permanent shutdown after which they SHOULD return a helpful http status code and text, for example `410 service no longer available ...` for all feed endpoint URLs.
```

**Pull request merged**